homeassistant/components/nextdns/button.py

```diff
@@ -0,0 +1,58 @@
+"""Support for the NextDNS service."""
+from __future__ import annotations
+
+from typing import cast
+
+from homeassistant.components.button import ButtonEntity, ButtonEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import NextDnsStatusUpdateCoordinator
+from .const import ATTR_STATUS, DOMAIN
+
+PARALLEL_UPDATES = 1
+
+CLEAR_LOGS_BUTTON = ButtonEntityDescription(
+    key="clear_logs",
+    name="{profile_name} Clear Logs",
+    entity_category=EntityCategory.CONFIG,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add aNextDNS entities from a config_entry."""
+    coordinator: NextDnsStatusUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        ATTR_STATUS
+    ]
+
+    buttons: list[NextDnsButton] = []
+    buttons.append(NextDnsButton(coordinator, CLEAR_LOGS_BUTTON))
+
+    async_add_entities(buttons)
+
+
+class NextDnsButton(CoordinatorEntity, ButtonEntity):
```

Comment:
Could you use the generic here instead?
If you also use the generic in the Sensor platform, you should not longer need to override the coordinator type hint on `NextDnsSensor` and you could drop line 276 there.

```suggestion
class NextDnsButton(CoordinatorEntity[NextDnsStatusUpdateCoordinator], ButtonEntity):
```

Reply:
@epenet I'll open a new PR to address this in a sensor platform, thanks.

Reply:
https://github.com/home-assistant/core/pull/74503